src/lib/input/input.scss

```diff
@@ -52,6 +52,9 @@
   &[type='month'],
   &[type='week'],
   &[type='time'] {
+    // Hack to prevent a line break on Safari input date fields:
+    white-space: pre;
```

Comment:
We should check if this is still necessary after switching back to the `::first-line` method

Reply:
Both of these look fine on desktop Safari (11.0.3). I'll see if I can steal someone's iPhone tomorrow to test mobile Safari. I'll lgtm for now, and add merge ready after we confirm if the hack is needed.

Reply:
Here are some screenshots taken by BrowserStack. I've reduced the input width in order to highlight the bug (look at the date field).

![bs_realios_mobile_iphone 6s-9 0 without pre reduced](https://user-images.githubusercontent.com/5031156/38123452-719a366c-33b1-11e8-801f-e64cc374d530.jpg)

![bs_realios_mobile_iphone 6s-9 0 with pre reduced](https://user-images.githubusercontent.com/5031156/38123451-71795c80-33b1-11e8-8c7d-58379b599fff.jpg)

Also, after a rebase I noticed that the duplicate label issue has returned. Should I add a fix in my PR? Or should I wait again for an external fix as before?

Reply:
PS.: returned for the Safari version running from BrowserStack. I've unable to reproduce in other browsers.

Reply:
A last comment: I could not replicate the error at build with Edge. I dont think that is related, but I want to ask you: is it possible that the problem is related to something in my PR? At least in my tests the input page is working on Edge.

Reply:
Its probably a flake, I've restarted the tests

Reply:
Confirmed that the hack is still needed for iPad

Reply:
It failed on Safari now oO'.

Thank you very much for your help and patience with this PR.

Reply:
Yeah, I'll just rerun it till it passes, definitely seems like flakiness. Same to you, thanks for sticking with this!

Reply:
@mmalerba Thanks. Are you available on Gitter?

Reply:
I've never actually signed in before, but I probably should. Just joined